scripts/dev/generate_config_types.py

```diff
@@ -0,0 +1,237 @@
+"""This script auto-generates the `qutebrowser/config/configcontainer.py` file."""
+
+from __future__ import annotations
+
+from pathlib import Path
+from textwrap import dedent
```

Comment:
nitpick: Except for type annotations (which benefit from being short), we typically import modules in qutebrowser:

```suggestion
import pathlib
import textwrap
```

That makes it clearer when reading the code where things are coming from, and that e.g. `dedent` isn't just a function defined inside this module.